gamemode/modules/fadmin/fadmin/cl_interface/cl_scoreboardServer.lua

```diff
@@ -56,23 +56,23 @@ local function MakeServerOptions()
     end
 
     for k, v in ipairs(FAdmin.ScoreBoard.Server.ActionButtons) do
-        local visible = v.Visible == true or (type(v.Visible) == "function" and v.Visible(LocalPlayer()) == true)
+        local visible = v.Visible == true or (isfunction(v.Visible) and v.Visible(LocalPlayer()) == true)
 
         local ActionButton = vgui.Create("FAdminActionButton")
-        if type(v.Image) == "string" then
+        if isstring(v.Image) then
```

Comment:
Since the type is checked in three different ways here, it would be more efficient to grab the TypeID once and compare it three times. TypeID is better than type since it's number comparison instead of strings.

Reply:
If I understand you correctly. I think it's unnecessary. The code should be readable.

Reply:
I think he wants you to use [this](https://wiki.garrysmod.com/page/Global/TypeID) instead `isstring` check

Reply:
> I think he wants you to use [this](https://wiki.garrysmod.com/page/Global/TypeID) instead `isstring` check

Yeah, that's what I figured. But I don't see any need for that. It only complicates the code's readability.

Reply:
Not really, it's just as readable but more efficient. Just name the variable ``imageType`` and there will be no confusion.

Reply:
> Not really, it's just as readable but more efficient. Just name the variable `imageType` and there will be no confusion.

It really makes reading difficult. If you compare, you will need more characters and more characters to declare a variable. It also requires more memory (which is not important). 

What we have now is the easiest option.

Reply:
Isn't the point of this PR efficiency? I don't see how character count is even a factor here.

Reply:
> Isn't the point of this PR efficiency? I don't see how character count is even a factor here.

You want to do that? Did I understand correctly?
```Lua
local ActionButton = vgui.Create("FAdminActionButton")
local imageType = TypeID(v.Image)
if imageType == 4 then -- string
    ActionButton:SetImage(v.Image or "icon16/exclamation")
elseif imageType == 5 then -- table
    ActionButton:SetImage(v.Image[1])
    if v.Image[2] then ActionButton:SetImage2(v.Image[2]) end
elseif imageType == 1 then -- function
```

Reply:
Also you can cache v.Image to a local instead of indexing it multiple times.